analyses/mwi/dryspells_trigger/exploration/2021_2022_postseason_data_analysis.R

```diff
@@ -0,0 +1,148 @@
+# analyse precipitation data for postseason analysis of 2021-2022 season
+# this script sources 2021_2022_postseason_data_pull.R
+# this script is sourced by 2021_2022_postseason_overview.Rmd
+
+#####
+## data pull
+#####
+
+# load data and useful R objects
+source('dryspells_trigger/exploration/2021_2022_postseason_data_pull.R')
+
+#####
+## data formatting
+#####
+
+# user-friendly date labels
+dates <- data.frame(dates = seq(from = as.Date('2021-10-01', format = '%Y-%m-%d'), 
+                   to = as.Date('2022-03-20', format = '%Y-%m-%d'), ## FIX ME
+                   # to = as.Date('2022-04-01', format = '%Y-%m-%d'), 
+                   by = "days"))
+
+date_labels <- cbind(dates_chr, dates)
+
+# pivot data to long format & formatting
+lf <- data %>% 
+        select(-ID, -x, -y) %>%
+        pivot_longer(
+          cols = starts_with("est_prcp_T="),
+          names_to = "date_chr",
+          values_to = "rainfall"
+          #values_drop_na = TRUE
+        ) %>%
+        left_join(date_labels, by = c('date_chr' = 'dates_chr')) %>%
+        mutate(month = lubridate::month(dates), # identify calendar month
+               week = lubridate::week(dates), # identify calendar week
+               rainy_bin = ifelse(rainfall >= 3, 1, 0)) %>% # binary: was >= 3mm of rain received? Rainy day definition from February 2022 bulletin from DCCMS
+      select(cell, date_chr, dates, month, week, rainfall, rainy_bin) %>%
+      arrange(cell, dates) 
+
+#####
+# analysis
+#####
+
+## 
+# Rainy season stats
+##
+
+# compute rolling sums
+stats <- lf %>%
+  group_by(cell) %>%
+  arrange(cell, dates) %>%
+  dplyr::mutate(roll_sum_next_10d = ifelse(!is.na(rainfall), zoo::rollsum(rainfall, k = 10, fill = NA, align = 'left'), NA), # sum of precipitation in 10-day period starting today
```

Comment:
`ifelse` here is redundant and can be removed.

Reply:
Oh yeah that's a relic from making the rollsum, lead/lag, etc work.